src/plugins/mer/meremulatormodedialog.cpp

```diff
@@ -159,18 +160,18 @@ void MerEmulatorModeDialog::execDialog()
 
     m_ui->deviceNameLabel->setText(m_emulator.data()->displayName());
 
-    const QMap<QString, MerEmulatorDeviceModel> models =
-        m_emulator.data()->availableDeviceModels();
-    const bool supportsMultipleModels = !models.isEmpty();
+    const QStringList names = MerSettings::deviceModelNames();
+    const bool supportsMultipleModels = !names.isEmpty();
```

Comment:
I would prefer following what I suggested here https://github.com/sailfishos/sailfish-qtcreator/pull/141#discussion_r239359327

> So there should be deviceModels()/addDeviceModel()/removeDeviceModel() in MerSettings and deviceModel()/setDeviceModel() in MerEmulatorDevice. MerSettings should save/read the list of deviceModels in save()/read(). MerEmulatorDevice should store the deviceModel().name in toMap() and restore deviceModel using the stored name in fromMap(). After that also the connection to devicesLoaded will not be needed.

I.e., MerSettings::deviceModels() should return the same type as the original MerEmulatorDevice::availableDeviceModels() does. With that approach the code below does not need to be changed.

Reply:
As I said, QMap doesn't fit if user is able to set profile names. The QMap sorting with name as a key may not work properly. That is why I introduce this change.

Reply:
ditto, if you really really think the QCollator usage is needed, please move that near the UI layer and keep the lower layers using QMap. I wouldn't use QHash so that the saved content is somehow stable.

Reply:
Done.